frontend/src/utils.js

```diff
@@ -0,0 +1,24 @@
+export async function fetchDataProduct(definition, params) {
+  // In this application we use data products that are published under
+  //  "ioxio" source only
+  try {
+    const resp = await fetch(`/api/data-product/${definition}?source=ioxio`, {
+      method: "POST",
+      body: JSON.stringify(params),
+    })
+    return await resp.json()
+  } catch (err) {
+    throw new Error("Failed to fetch a data product")
+  }
+}
+
+export async function getUser() {
+  try {
+    const resp = await fetch(`/api/me`, {
+      method: "GET",
+    })
+    return await resp.json()
+  } catch (err) {
+    throw new Error("Failed to fetch a user")
```

Comment:
```suggestion
    throw new Error("Failed to fetch the current user")
```

Reply:
Not what I meant. This is now first catching an exception, then throwing another one. So adding absolutely nothing of value.

I said `if (!resp.ok) throw ...`